### ISSUE TITLE
Reduction of the Learning rate isn't working yet

### ISSUE THREAD

**Issue by beckstev** [beckstev/MachineLearningSeminar#10]
We should implement in future version a dynamic leraning rate. See the following post: 

https://github.com/keras-team/keras/issues/2823
https://github.com/keras-team/keras/issues/898

**Comment by beckstev** [comment]
I implemented EarlyStopping and AdaptiveLeariningRate. 
We should also check out [Keras Callback](https://keras.io/callbacks/) functions.

**Comment by beckstev** [comment]
So I trained `DogNNv2` for 2.5 hours and you can see the loss is __decreasing__ :-1: 
I noticed that I forgot to activate `EarlyStopping` and `LRreducer` :-1: 

 
```
720/720 [==============================] - 533s 740ms/step - loss: 4.7865 - acc: 0.0102 - val_loss: 4.7853 - val_acc: 0.0124
Epoch 2/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7828 - acc: 0.0109 - val_loss: 4.7845 - val_acc: 0.0124
Epoch 3/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7806 - acc: 0.0115 - val_loss: 4.7846 - val_acc: 0.0122
Epoch 4/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7798 - acc: 0.0119 - val_loss: 4.7850 - val_acc: 0.0124
Epoch 5/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7794 - acc: 0.0114 - val_loss: 4.7854 - val_acc: 0.0122
Epoch 6/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7791 - acc: 0.0109 - val_loss: 4.7858 - val_acc: 0.0122
Epoch 7/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7790 - acc: 0.0115 - val_loss: 4.7854 - val_acc: 0.0122
Epoch 8/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7790 - acc: 0.0117 - val_loss: 4.7857 - val_acc: 0.0122
Epoch 9/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7789 - acc: 0.0120 - val_loss: 4.7861 - val_acc: 0.0122
Epoch 10/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7789 - acc: 0.0120 - val_loss: 4.7861 - val_acc: 0.0122
Epoch 11/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7789 - acc: 0.0115 - val_loss: 4.7860 - val_acc: 0.0122
Epoch 12/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7788 - acc: 0.0120 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 13/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7789 - acc: 0.0111 - val_loss: 4.7861 - val_acc: 0.0122
Epoch 14/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7788 - acc: 0.0120 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 15/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7788 - acc: 0.0117 - val_loss: 4.7862 - val_acc: 0.0122
Epoch 16/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7787 - acc: 0.0120 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 17/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7787 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 18/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7787 - acc: 0.0116 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 19/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7787 - acc: 0.0112 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 20/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7787 - acc: 0.0115 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 21/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7787 - acc: 0.0120 - val_loss: 4.7862 - val_acc: 0.0122
Epoch 22/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7787 - acc: 0.0120 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 23/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0110 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 24/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0099 - val_loss: 4.7862 - val_acc: 0.0122
Epoch 25/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0113 - val_loss: 4.7862 - val_acc: 0.0122
Epoch 26/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0117 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 27/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0116 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 28/10000
720/720 [==============================] - 94s 130ms/step - loss: 4.7786 - acc: 0.0120 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 29/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0120 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 30/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0115 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 31/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0110 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 32/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0120 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 33/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7786 - acc: 0.0120 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 34/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0105 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 35/10000
720/720 [==============================] - 94s 130ms/step - loss: 4.7785 - acc: 0.0116 - val_loss: 4.7863 - val_acc: 0.0122
Epoch 36/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0117 - val_loss: 4.7864 - val_acc: 0.0122
Epoch 37/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7786 - acc: 0.0117 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 38/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0109 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 39/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0115 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 40/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0117 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 41/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7785 - acc: 0.0116 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 42/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 43/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 44/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 45/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 46/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0109 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 47/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0111 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 48/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 49/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0104 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 50/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 51/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0117 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 52/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 53/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0114 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 54/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0106 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 55/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0111 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 56/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0109 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 57/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 58/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0111 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 59/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 60/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0111 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 61/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 62/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0119 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 63/10000
720/720 [==============================] - 95s 131ms/step - loss: 4.7785 - acc: 0.0108 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 64/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0115 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 65/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 66/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 67/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 68/10000
720/720 [==============================] - 94s 130ms/step - loss: 4.7785 - acc: 0.0109 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 69/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 70/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0115 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 71/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0115 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 72/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0116 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 73/10000
720/720 [==============================] - 94s 130ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 74/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0113 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 75/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0114 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 76/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 77/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0116 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 78/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 79/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 80/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 81/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0113 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 82/10000
720/720 [==============================] - 93s 130ms/step - loss: 4.7785 - acc: 0.0116 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 83/10000
720/720 [==============================] - 94s 130ms/step - loss: 4.7785 - acc: 0.0112 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 84/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0112 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 85/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0114 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 86/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7865 - val_acc: 0.0122
Epoch 87/10000
720/720 [==============================] - 95s 132ms/step - loss: 4.7785 - acc: 0.0115 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 88/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 89/10000
720/720 [==============================] - 94s 130ms/step - loss: 4.7785 - acc: 0.0114 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 90/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 91/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0108 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 92/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0120 - val_loss: 4.7866 - val_acc: 0.0122
Epoch 93/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0119 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 94/10000
720/720 [==============================] - 94s 131ms/step - loss: 4.7785 - acc: 0.0113 - val_loss: 4.7867 - val_acc: 0.0122
Epoch 95/10000
```

**Comment by beckstev** [comment]
Implemented a functional solution in https://github.com/beckstev/MachineLearningSeminar/pull/13
